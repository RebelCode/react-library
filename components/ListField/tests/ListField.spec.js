import React from "react";
import { render, cleanup } from "@testing-library/react";
import {ListField} from "../index";

afterEach(cleanup);

describe("ListField", () => {
  const Comp = (
    <ListField value={["Hakonan", "shoe"]} />
  );

  it("should take a snapshot", () => {
    const { asFragment } = render(Comp);

    expect(asFragment(Comp)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(Comp);

    expect(queryByLabelText("ListField")).toBeTruthy();
  });
});
