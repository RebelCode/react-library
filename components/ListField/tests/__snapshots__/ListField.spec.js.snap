// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListField should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="react-select css-dyr6gj-container"
  >
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class="react-select__control css-iq6f7g-control"
    >
      <div
        class="react-select__value-container react-select__value-container--is-multi react-select__value-container--has-value css-1vglxg7-ValueContainer"
      >
        <div
          class="css-1hgkiop-multiValue react-select__multi-value"
        >
          <div
            class="css-12jo7m5 react-select__multi-value__label"
          >
            Hakonan
          </div>
          <div
            class="css-xb97g8 react-select__multi-value__remove"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="14"
              viewBox="0 0 20 20"
              width="14"
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
        </div>
        <div
          class="css-1hgkiop-multiValue react-select__multi-value"
        >
          <div
            class="css-12jo7m5 react-select__multi-value__label"
          >
            shoe
          </div>
          <div
            class="css-xb97g8 react-select__multi-value__remove"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="14"
              viewBox="0 0 20 20"
              width="14"
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
        </div>
        <div
          class="css-1epxsz4-Input"
        >
          <div
            class="react-select__input"
            style="display: inline-block;"
          >
            <input
              aria-autocomplete="list"
              aria-label="ListField"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              id="react-select-2-input"
              spellcheck="false"
              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
          </div>
        </div>
      </div>
      <div
        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
      />
    </div>
  </div>
</DocumentFragment>
`;
